pkg/bpf/ops_linux.go

```diff
@@ -45,22 +45,17 @@ func (m StructBinaryMarshaler) MarshalBinary() ([]byte, error) {
 }
 
 type mapOps[KV KeyValue] struct {
-	m *ebpf.Map
-}
-
-func NewMapOps2[KV KeyValue](m *ebpf.Map) reconciler.Operations[KV] {
-	ops := &mapOps[KV]{m}
-	return ops
+	m *Map
 }
 
 func NewMapOps[KV KeyValue](m *Map) reconciler.Operations[KV] {
-	ops := &mapOps[KV]{m.m}
+	ops := &mapOps[KV]{m}
 	return ops
 }
 
 // Delete implements reconciler.Operations.
 func (ops *mapOps[KV]) Delete(ctx context.Context, txn statedb.ReadTxn, entry KV) error {
-	return ops.m.Delete(entry.BinaryKey())
+	return ops.m.m.Delete(entry.BinaryKey())
```

Comment:
Good point, this is now putting the responsibility on the user of NewMapOps to actually take care that the map is opened. MapOps should at the very least validate that this is the case and panic with a clear explanation, or it can keep trying until the map is opened. I'll think about this next week when I'm back...